feat: add tsoa for API type control and documentation (#1501)
finos/git-proxy: website/docs/architecture/architecture.md

If `apiAuthentication` is left empty, API endpoints will be publicly accesible.

> Currently, only JWT auth is supported. This is implemented via the [`@Security('jwt')` decorator](https://github.com/finos/git-proxy/blob/main/src/service/authentication.ts). Aside of validating incoming access tokens, it can also assign roles based on the token payload.

Reviewer: typo (Aside _of_ ...), Can improve wording slightly:

```suggestion
Currently, only JWT auth is supported. This is implemented via the [`@Security('jwt')` decorator](https://github.com/finos/git-proxy/blob/main/src/service/authentication.ts). In addition to validating incoming access tokens, the decorator assigns roles based on the token payload.
```
